### PR TITLE
Adding arguments to the install script

### PR DESCRIPTION
**Pull request by cintiadr**
I'd like to do unattended installations of docker using this script. 
So, I'd like it to: not wait if the 'docker' command is there, also it should just exit successfully. 

So I added two arguments to the script. 

(I don't have any non-bash shell to test, so I might have made a few bashisms by mistake)

Tested on centos 6.5, 6.6, ubuntu 12 and 14.

### CODE REVIEW

File: hack/install.sh [hunk 1]
```diff
@@ -19,11 +21,38 @@ set -e
...
+	EOF
+}
+
+parse_arguments(){
+        while getopts ":hw:r:" opt; do
+                case "$opt" in
+                        h)  show_help
+                            exit 0
+                            ;;
+                        w)  if  [ "$OPTARG" == "false" ]; then wait_if_docker_command_exists=0 ; fi
```
Comment (cintiadr) [inline hack/install.sh]: Look. I don't like this IF. But anything else I've tried ended up in the 0-or-1 is success trap, which just made the code insane to follow.

File: hack/install.sh [hunk 2]
```diff
@@ -5,6 +5,8 @@ set -e
 #   'curl -sSL https://get.docker.com/ | sh'
 # or:
 #   'wget -qO- https://get.docker.com/ | sh'
+# If you want to send arguments:
+#   'sh <(wget -qO- https://get.docker.com/) -w false -r true'
```
Comment (cintiadr) [inline hack/install.sh]: Is this only for bash?

**Comment by cintiadr** [comment]
I don't think this change would break the experimental build...

**Comment by icecrime** [comment]
Ping @tianon!

**Comment by jfrazelle** [comment]
hmmm the idea seems fine to me, just more for us to maintain as far as functionality.. do people really curl | sh to update something?

**Comment by jfrazelle** [comment]
just not sure if we should be condoning such behavior ;)

**Comment by thaJeztah** [comment]
Wondering if docker machine would be a better solution for automated installs here

**Comment by cintiadr** [comment]
@jfrazelle If people actually use curl | sh to update I'm not sure. But that's pretty much keeping the old behaviour of the script, but I'm happy to remove it (if it's not considered a breaking change). 

@thaJeztah I believe we shouldn't be using docker machines for prod yet?

**Comment by thaJeztah** [comment]
> I believe we shouldn't be using docker machines for prod yet?

Yeah, you're right on that. Otoh, using this script to automatically (attempt to) detect the platform I'm running on, and automatically install the latest release of Docker may not be what you want in production either; in production, I *know* the platform I'm running on, and I want to explicitly specify the version of Docker that will be installed (otherwise, I may get unpleasant surprises, if one of my machines all of a sudden runs a new version that I haven't tested on)

**Comment by cpuguy83** [comment]
Sounds like we would prefer to not have this?
Should we close?

**Comment by jfrazelle** [comment]
I think we should, I'm sorry thank you so much for the patch but I do not think this is something we want people to rely on at this time

**Pull request closed without merging**